tests/providers/yandex/secrets/test_lockbox.py

```diff
@@ -63,11 +65,15 @@ def test_yandex_lockbox_secret_backend_get_connection_from_json(self, mock_get_v
 
         mock_get_value.return_value = json.dumps(c)
 
-        conn = LockboxSecretBackend().get_connection(conn_id)
+        with pytest.warns(
+            RemovedInAirflow3Warning,
+            match="Encountered non-JSON in `extra` field for connection 'airflow_to_yandexcloud'. Support for non-JSON `extra` will be removed in Airflow 3.0",
+        ):
+            conn = LockboxSecretBackend().get_connection(conn_id)
 
-        assert conn.conn_id == conn_id
-        assert conn.conn_type == conn_type
-        assert conn.extra == extra
+            assert conn.conn_id == conn_id
+            assert conn.conn_type == conn_type
+            assert conn.extra == extra
```

Comment:
I'd probably change to something like this to isolate the causes of the warning:
```python
        match = "Encountered non-JSON in `extra` field for connection 'airflow_to_yandexcloud'."
        with pytest.warns(
            RemovedInAirflow3Warning,
            match=match,
        ):
            conn = LockboxSecretBackend().get_connection(conn_id)

        with pytest.warns(
            RemovedInAirflow3Warning,
            match=match,
        ):
            assert conn.extra == extra

        assert conn.conn_id == conn_id
        assert conn.conn_type == conn_type
```